frontend/src/config/configure.ts

```diff
@@ -0,0 +1,153 @@
+import path from 'path';
+
+import type { BeamerConfig } from '@/types/config';
+
+import type { ChainMetadata } from './chains/chain';
+import type { DeploymentInfo } from './deployment';
+import { TokenMetadata } from './tokens/token';
+import type { BeamerConfigEnvMapping, Environment } from './types';
+import {
+  getEnvironmentForFolder,
+  readAppConfigFile,
+  readChainMetadataFolder,
+  readDeploymentFolder,
+  readTokenMetadataFolder,
+  writeAppConfigFile,
+} from './utils';
+
+const CHAINS_DIR = path.join(__dirname, 'chains');
+const TOKENS_DIR = path.join(__dirname, 'tokens');
+const DEPLOYMENT_DIR = path.join(__dirname, '../../../deployments');
+const OUTPUT_DIR = path.join(__dirname, '../../public');
+
+export const generateTestToken = (rootDeploymentFolder?: string) => {
+  const TST = new TokenMetadata({
+    symbol: 'TST',
+    decimals: 18,
+    addresses: {},
+  });
+
+  const deploymentInfos = readDeploymentFolder(rootDeploymentFolder ?? DEPLOYMENT_DIR, [
+    'mainnet',
+  ]);
+
+  const chainMintableTokenAddressMapping = {};
+
+  for (const info of deploymentInfos)
+    Object.assign(chainMintableTokenAddressMapping, info.getMintableTokenAddresses());
+
+  TST.addAddresses(chainMintableTokenAddressMapping);
+
+  return TST;
+};
+
+/**
+ * Mixes deployment & metadata info
+ */
+export const generateAppConfig = (
+  deploymentInfo: DeploymentInfo,
+  chainMetas: ChainMetadata[],
+  tokenMetas: TokenMetadata[],
+): BeamerConfig => {
+  const config = { chains: {} };
+
+  const chainWithTokens = chainMetas.map((chainMeta) =>
+    chainMeta.formatUsingTokenMetas(tokenMetas),
+  );
+
+  for (const chainId of deploymentInfo.supportedChains) {
+    const chainMeta = chainWithTokens.find((chain) => chain.identifier == Number(chainId));
+    // Avoid mixing chains that have no metadata
+    if (!chainMeta) continue;
+
+    Object.assign(config.chains, {
+      [chainId]: {
+        ...chainMeta,
+        ...deploymentInfo.formatChainDeploymentInfo(chainId),
+      },
+    });
+  }
+  return config;
+};
+
+class ConfigPropertyValueError extends Error {
+  constructor(filePath: string, property: string) {
+    super(`[${filePath}]: Configuration corrupted. Invalid '${property}' value`);
+    Object.setPrototypeOf(this, ConfigPropertyValueError.prototype);
```

Comment:
Its only used inside the `verifyAppConfigFile` function. 
Basically a "throwable" error that is related to the file verification process which holds information about the file path and the exact property where the type error was spotted.

Reply:
I just meant this function call: `Object.setPrototypeOf(this, ConfigPropertyValueError.prototype);`
I still don't get this line.

Reply:
Honestly, i copied this throwable implementation from [here](https://github.com/beamer-bridge/beamer/blob/main/frontend/src/services/transactions/request-manager.ts#L223-L228).

Maybe this can answer your question https://stackoverflow.com/a/45043131